assets/src/edit-story/output/utils/getFontDeclarations.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import getGoogleFontURL from './getGoogleFontURL';
+
+/**
+ * Returns a list of font declarations across all pages in a story.
+ *
+ * @param {import('../../types').Page[]} pages List of pages.
+ *
+ * @return {string[]} Font declarations.
+ */
+const getFontDeclarations = (pages) => {
+  const map = new Map();
+
+  for (const { elements } of pages) {
+    const textElements = elements.filter(({ type }) => type === 'text');
+
+    // Prepare font objects for later use.
+    for (const { font, fontStyle, fontWeight } of textElements) {
```

Comment:
Do we also need (now or later) to estimate the unicode character sets used? E.g. `?subset=X` in the URL? TBH, I don't know how much subset helps/hurts in reality.

Reply:
The docs say:

> The Latin subset is always included if available and need not be specified. Please note that if a client browser supports unicode-range (http://caniuse.com/#feat=font-unicode-range) the subset parameter is ignored; the browser will select from the subsets supported by the font to get what it needs to render the text.

The browser support there matches with what we cover, so I don't think we need to do anything about subsets.